lib/ecto/type.ex

```diff
@@ -860,7 +860,7 @@ defmodule Ecto.Type do
   # Checks if a value is of the given primitive type.
   defp of_base_type?(:any, _),         do: true
   defp of_base_type?(:id, term),       do: is_integer(term)
-  defp of_base_type?(:float, term),    do: is_float(term)
+  defp of_base_type?(:float, term),    do: is_float(term) or is_integer(term)
```

Comment:
I don't like this change because it means we would be able to write `%{measurement | intensity: some_integer}` even when intensity is tagged as `float`. What happens if we remove this?

One option is to try to mark all of them as decimal because decimal accept integers, floats and even binaries, mirroring closer to the database behaviour.

Reply:
Sorry, I didn't see this comment. Without this, I can't use integers in a operation that requires a float, like `type(p.views / 2, :float)` (`p.views / 2` works just fine IIRC).

My reasoning behind this change is that although you can't use a float where an integer is expected, the only operation I can think off where the opposite is true is the `is_float/1` function.

So I didn't see as much of an issue allowing integers where floats are expected. Thoughts?